src/raster/rasterband.rs

```diff
@@ -612,3 +621,144 @@ impl ColorInterpretation {
         _string(rv)
     }
 }
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub enum PaletteInterpretation {
+    Gray,
+    Rgba,
+    Cmyk,
+    Hls,
+}
+
+impl PaletteInterpretation {
+    fn from_c_int(palette_interpretation: GDALPaletteInterp::Type) -> Self {
+        match palette_interpretation {
+            GDALPaletteInterp::GPI_Gray => Self::Gray,
+            GDALPaletteInterp::GPI_RGB => Self::Rgba,
+            GDALPaletteInterp::GPI_CMYK => Self::Cmyk,
+            GDALPaletteInterp::GPI_HLS => Self::Hls,
+            _ => unreachable!("GDAL has implemented a new type of `GDALPaletteInterp`"),
+        }
+    }
+}
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub struct GrayEntry {
+    pub g: i16,
+}
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub struct RgbaEntry {
+    pub r: i16,
+    pub g: i16,
+    pub b: i16,
+    pub a: i16,
+}
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub struct CmykEntry {
+    pub c: i16,
+    pub m: i16,
+    pub y: i16,
+    pub k: i16,
+}
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub struct HlsEntry {
+    pub h: i16,
+    pub l: i16,
+    pub s: i16,
+}
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub enum ColorEntry {
+    Gray(GrayEntry),
+    Rgba(RgbaEntry),
+    Cmyk(CmykEntry),
+    Hls(HlsEntry),
+}
+
+/// Color table for raster bands that use the PaletteIndex color interpretation.
+///
+/// This object carries the lifetime of the raster band that
+/// contains it. This is necessary to prevent the raster band
+/// from being dropped before the color table.
+pub struct ColorTable<'a> {
+    palette_interpretation: PaletteInterpretation,
+    c_color_table: GDALColorTableH,
+    _raster_band: &'a RasterBand<'a>,
```

Comment:
Can this be a `PhantomData<&'a RasterBand<'a>>` or something?

Reply:
I was unaware of `PhantomData`. Thanks for pointing it out.